In [2]:
library(tidyverse)
library(repr)
source("cleanup.R")
options(repr.matrix.max.rows = 6)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Warning message in file(filename, "r", encoding = encoding):
“cannot open file 'cleanup.R': No such file or directory”


ERROR: Error in file(filename, "r", encoding = encoding): cannot open the connection


In [30]:
# read_players_file <- read_csv("/home/jovyan/work/DSCI100_Individual_Project/data/players.csv")
# read_players_file <- read_players_file |>
#   mutate(
#     experience = factor(
#       experience,
#       levels = c("Beginner", "Amateur", "Regular", "Veteran", "Pro"),
#       ordered = TRUE
#     ), gender = factor(gender,levels = c("Male","Female","Non-binary","Prefer not to say","Agender","Two-Spirited","Other"),ordered = TRUE)
#   )
# read_players_file|>
#   summarize(
#     mean_age = mean(Age, na.rm = TRUE),
#     min_age = min(Age, na.rm = TRUE),
#     max_age = max(Age, na.rm = TRUE),
#     mean_hours = mean(played_hours, na.rm = TRUE),
#     min_hours = min(played_hours, na.rm = TRUE),
#     max_hours = max(played_hours, na.rm = TRUE),
#   )
# read_players_file |>
#   count(experience, sort = TRUE) |>
#   slice(1)
# read_players_file |>
#   count(subscribe, sort = TRUE) |>
#   slice(1)
# read_players_file |>
#   count(gender, sort = TRUE)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


mean_age min_age max_age mean_hours min_hours max_hours
1 21.13918 9       58      5.845918   0         223.1

experience n 
1 Amateur    63

subscribe n  
1 TRUE      144

gender            n  
1 Male              124
2 Female             37
3 Non-binary         15
4 Prefer not to say  11
5 Two-Spirited        6
6 Agender             2
7 Other               1

***Data Description and Statistics of Players.csv file***

The number of observations in the players file is ***196 players*** and ***7 columns (headings)***. The name and description of variables are as follows: 
1. Experience: This variable records the level of experience that each player reports with the game. It is a categorical variable with five categories: Beginner, Amateur, Regular, Veteran, and Pro. The most common group is Amateur, with 63 players.

2. Subscribe: Indicates whether the player has subscribed to the research newsletter. It is a categorical variable with values TRUE or FALSE. Out of 196 players, 144 have subscribed.

3. HashedEmail: A character variable representing each player’s unique ID.

4. Played_hours: Shows the total number of hours a player has spent on the server. It is a numeric variable. The average number of hours played is 5.85, with a minimum of 0 and a maximum of 223.10 hours.

5. Name: A character variable storing the in-game display name chosen by each player. Every name is unique. 

6. Gender: A categorical variable describing the player’s gender. There are 7 categories (male, female, agender, other, two-spirited, prefer not to say, and non-binary), with male being the most frequent (124 players).

7. Age: A numeric variable representing the player’s age in years. The average age is 21.14, with a minimum age of 9 and a maximum age of 58 years old.


The ***Data*** statstics collected using the formula of R and it is shown above.

The data came from the research group in computer science at UBC, which is led by Frank Wood. They collected the data by setting up the Minecraft server, and the actions are recorded as they go through the world. Two datasets are provided, which will capture players' demographics and engagement in the game namely ***"players.csv"***.

Yes, there are some issues with the dataset, such as the Gender, Subscribe, and Experience columns, which should be categorical variables, not character variables. This can cause incorrect statistics, specifically when we want to use max and min functions using R. Moreover, there are two missing values in the age column.

Also, there are some other potentials that can't be seen as of now, such as it being a voluntary dataset, not a random dataset, and from the class of statistics, we have learned that we can't generalize these results to the larger audience and make inferences out of it. There are some players who have joined the Minecraft server but never played on the server, and these can impact the results of the statistics.

The number of observations in the session file contains 1535 rows (play sessions) and 5 columns (headings).